apps/test/unit/lib/levelbuilder/lesson-editor/ObjectivesEditorTest.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {expect} from '../../../../util/reconfiguredChai';
+import ObjectivesEditor from '@cdo/apps/lib/levelbuilder/lesson-editor/ObjectivesEditor';
+
+describe('ObjectivesEditor', () => {
+  let defaultProps;
+  beforeEach(() => {
+    defaultProps = {
+      objectives: [{key: '1', description: 'description'}]
+    };
+  });
+
+  it('renders default props', () => {
+    const wrapper = shallow(<ObjectivesEditor {...defaultProps} />);
+    expect(wrapper.find('tr').length).to.equal(2);
+  });
+
+  it('can remove a objective', () => {
+    const wrapper = shallow(<ObjectivesEditor {...defaultProps} />);
+    const numObjectives = wrapper.find('tr').length;
+    expect(numObjectives).at.least(2);
+    // Find one of thet "remove" buttons and click it
+    const removeObjectiveButton = wrapper
+      .find('.fa-trash')
+      .first()
+      .parent();
+    removeObjectiveButton.simulate('mouseDown');
+    expect(wrapper.find('tr').length).to.equal(numObjectives - 1);
+  });
+
+  it('can add a objective', () => {
+    const wrapper = shallow(<ObjectivesEditor {...defaultProps} />);
+    const numObjectives = wrapper.find('tr').length;
+    const addObjectiveButton = wrapper.find('button').first();
+    addObjectiveButton.simulate('mouseDown');
+    const objectiveInput = wrapper.find('input[type="text"]').first();
+    objectiveInput.simulate('change', {target: {value: 'new description'}});
+    objectiveInput.simulate('keyDown', {key: 'Enter'});
+    expect(wrapper.find('tr').length).to.equal(numObjectives + 1);
+  });
+
+  it('does not add an empty objective', () => {
+    const wrapper = shallow(<ObjectivesEditor {...defaultProps} />);
+    const numObjectives = wrapper.find('tr').length;
+    const addObjectiveButton = wrapper.find('button').first();
+    addObjectiveButton.simulate('mouseDown');
+    const cancelEditButton = wrapper
+      .find('.fa-times')
+      .first()
+      .parent();
+    cancelEditButton.simulate('mouseDown');
+    expect(wrapper.find('tr').length).to.equal(numObjectives);
+  });
```

Comment:
I think I get why you named the test case this, but the name is making me think that you're trying to reject empty objective when save is pressed whereas it looks like you are actually testing what happens when cancel is pressed. could we make the name a bit more explicit like: "does not add an objective after add and cancel" or something like that?

Reply:
Oh hm this is making me realize my empty objective/cancel logic is off.

I think what I want it to be is:
- If the user adds an objective then hits cancel, that objective isn't saved, no matter whether they typed anything or not
- If the user adds an objective, types nothing, then hits save, a blank objective should be saved. (Obviously, if something is typed then the user hits save, it should also be saved.)

I'll rename this test and then also fix the bug in `ObjectiveEditor` around this.